src/Drivers/check_determinant.cpp

```diff
@@ -0,0 +1,203 @@
+    case 'v': verbose = true; break;
+    case 'V':
+      print_version(true);
+      return 1;
+      break;
+    }
+  }
+
+  Random.init(0, 1, iseed);
+  Tensor<int, 3> tmat(na, 0, 0, 0, nb, 0, 0, 0, nc);
+
+  print_version(verbose);
+
+  // turn off output
+  if (!verbose || omp_get_max_threads() > 1)
+  {
+    OhmmsInfo::Log->turnoff();
+    OhmmsInfo::Warn->turnoff();
+  }
+
+  Tensor<OHMMS_PRECISION, 3> lattice_b;
+  ParticleSet ions, els;
+  OHMMS_PRECISION scale = 1.0;
+  lattice_b             = tile_cell(ions, tmat, scale);
+  ions.setName("ion");
+  els.setName("e");
+
+  // create generator within the thread
+  RandomGenerator<RealType> random_th(myPrimes[0]);
+
+  ions.Lattice.BoxBConds = 1;
+  ions.RSoA              = ions.R; // fill the SoA
+
+  const int nels  = count_electrons(ions, 1);
+  const int nels3 = 3 * nels;
+
+  { // create up/down electrons
+    els.Lattice.BoxBConds = 1;
+    els.Lattice.set(ions.Lattice);
+    vector<int> ud(2);
+    ud[0] = nels / 2;
+    ud[1] = nels - ud[0];
+    els.create(ud);
+    els.R.InUnit = 1;
+    random_th.generate_uniform(&els.R[0][0], nels3);
+    els.convert2Cart(els.R); // convert to Cartiesian
+    els.RSoA = els.R;
+  }
+
+  miniqmcreference::DiracDeterminantRef determinant_ref(nels, random_th);
+  DiracDeterminant determinant(nels, random_th);
+
+  // For VMC, tau is large and should result in an acceptance ratio of roughly
+  // 50%
+  // For DMC, tau is small and should result in an acceptance ratio of 99%
+  const RealType tau = 2.0;
+
+  ParticlePos_t delta(nels);
+
+  RealType sqrttau = std::sqrt(tau);
+  RealType accept  = 0.5;
+
+  aligned_vector<RealType> ur(nels);
+  random_th.generate_uniform(ur.data(), nels);
+
+  els.update();
+
+  int my_accepted = 0;
+  for (int mc = 0; mc < nsteps; ++mc)
+  {
+    determinant_ref.recompute();
+    determinant.recompute();
+    for (int l = 0; l < nsubsteps; ++l) // drift-and-diffusion
+    {
+      random_th.generate_normal(&delta[0][0], nels3);
+      for (int iel = 0; iel < nels; ++iel)
+      {
+        // Operate on electron with index iel
+        els.setActive(iel);
+
+        // Construct trial move
+        PosType dr   = sqrttau * delta[iel];
+        bool isValid = els.makeMoveAndCheck(iel, dr);
+
+        if (!isValid) continue;
+
+        // Compute gradient at the trial position
+
+        determinant_ref.ratio(iel);
+        determinant.ratio(iel);
+
+        // Accept/reject the trial move
+        if (ur[iel] > accept) // MC
+        {
+          // Update position, and update temporary storage
+          els.acceptMove(iel);
+          determinant_ref.acceptMove(iel);
+          determinant.acceptMove(iel);
+          my_accepted++;
+        }
+        else
+        {
+          els.rejectMove(iel);
+        }
+      } // iel
+    }   // substeps
+
+    els.donePbyP();
+  }
+
+  int errors = 0;
+  for (int i = 0; i < determinant_ref.size(); i++)
+  {
+    if (determinant_ref(i) != determinant(i)) ++errors;
```

Comment:
There should be tolerance on this check.  
In addition to printing out number of elements that fail, the accumulated error should be printed.  This would help with a quick diagnostic (numerical error just past tolerance, or a clear bug).

Reply:
Okay, I agree. This was a quick hack to get something up for review, and forgot to improve it.